docarray/array/doc_dict/doc_dict.py

```diff
@@ -0,0 +1,88 @@
+from typing import TYPE_CHECKING, Any, Dict, Mapping, Type, TypeVar, Union
+
+from typing_inspect import is_union_type
+
+from docarray.array.any_collections import AnyCollections
+from docarray.array.doc_list.doc_list import DocList
+from docarray.base_doc import AnyDoc
+from docarray.base_doc.doc import BaseDoc
+
+if TYPE_CHECKING:
+    from docarray.typing import NdArray, TorchTensor
+
+T_doc = TypeVar('T_doc', bound=BaseDoc)
+T = TypeVar('T', bound='DocDict')
+
+
+class DocDict(AnyCollections[T_doc], Dict[str, T_doc]):
+    doc_type: Type[BaseDoc] = AnyDoc
+
+    def __init__(self, **mapping):
+        super().__init__(
+            {key: self._validate_one_doc(val) for key, val in mapping.items()}
+        )
+
+    def _validate_one_doc(self, doc: T_doc) -> T_doc:
+        """Validate if a Document is compatible with this `DocList`"""
+        if not issubclass(self.doc_type, AnyDoc) and not isinstance(doc, self.doc_type):
+            raise ValueError(f'{doc} is not a {self.doc_type}')
+        return doc
+
+    @classmethod
+    def from_doc_list(cls: Type[T], docs: DocList) -> T:
+        return cls(**{doc.id: doc for doc in docs})
+
+    # here we need to ignore type as the DocDict as a signature incompatible with Dict (it is more restrictive)
+    def update(self, other: Union['DocDict', Dict[str, T_doc], DocList]):  # type: ignore
```

Comment:
does `Sequence[Doc]` not work, only `DocList`?
Can't we in that case just do `super().update(DocDict.from_doc_list(DocList[self.doc_type](other)))`?